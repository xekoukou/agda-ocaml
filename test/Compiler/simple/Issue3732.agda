open import Common.IO
open import Common.Unit
open import Common.String

-- Currently, it is not actually a test.
-- I need a wat to check that Erasure does not happen when it normally would.

{-# FOREIGN OCaml 

type i = 
  | Bar of string;;

#-}


data I : Set where
  bar : String → I

{-# COMPILE OCaml I No-Erasure #-}


showI : I → String
showI (bar x) = x

main : IO Unit
main = putStr (showI (bar "hello")) >>= λ _ → return unit
