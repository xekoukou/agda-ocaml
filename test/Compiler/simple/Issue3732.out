EXECUTED_PROGRAM

ret > ExitSuccess
out > hello
